runtime/compiler/env/VMJ9.cpp

```diff
@@ -8607,7 +8645,22 @@ TR_J9SharedCacheVM::getInstanceFieldOffset(TR_OpaqueClassBlock * classPointer, c
    TR::Compilation* comp = _compInfo->getCompInfoForCompOnAppThread() ? _compInfo->getCompInfoForCompOnAppThread()->getCompilation() : _compInfoPT->getCompilation();
    TR_ASSERT(comp, "Should be called only within a compilation");
 
-   if (((TR_ResolvedRelocatableJ9Method*) comp->getCurrentMethod())->validateArbitraryClass(comp, (J9Class*)classPointer))
+   bool validated = false;
+
+   if (comp->getOption(TR_UseSymbolValidationManager))
+      {
+      if (comp->getSymbolValidationManager()->inHeuristicRegion() ||
```

Comment:
when does one know to insert these `inHeuristicRegion()` calls, since they seem to be in some places but not others? It's a shame we couldn't find a way to sink them all inside the SVM or hide them behind existing queries.

Reply:
it seems to be mostly in places where it's just trying to validate that a class has already been validated....maybe the symbol validation manager needs a `verifySymbolHasID(void *)` service to avoid exposing the heuristic region stuff?

Reply:
> maybe the symbol validation manager needs a verifySymbolHasID(void *) service to avoid exposing the heuristic region stuff

Yeah that would be much cleaner